### PR TITLE
Upgrade to Go 1.19

### PR DESCRIPTION
**Pull request by dannykopping**
**What this PR does / why we need it**:
This PR upgrades Loki and all related components to Go 1.19.
Go 1.19 was released on August 2nd (almost 2 months from the time of writing), and has received 1 patch release thus far (1.19.1, release September 6th, almost 1 month from the time of writing). This gives me enough confidence that this version is solid and worth upgrading to.

_Update: since opening this PR, a security release (Go 1.19.2) was released - and I've used that instead. It's risky because it just came out yesterday, but due to the nature of the security fixes I think it's justified._

**Upgrade Checklist**:
- [x] Update go.mod and run `go mod vendor && go mod tidy`
- [x] Update `.golangci.yml` language version
- [x] Update Dockerfiles
- [x] [Update loki-build-image](https://github.com/grafana/loki/blob/main/docs/sources/maintaining/release-loki-build-image.md) (https://github.com/grafana/loki/pull/7334) including `golangci-lint` (https://github.com/grafana/loki/pull/7337)
- [ ] Create subsequent PR to update GEL

**Note:** I have disabled some irritating/noisy linters for now in order to get this merged.
I will restore them and fix the "issues" in a subsequent PR to not make this one too huge to review carefully.

### CODE REVIEW

**Review by periklis — COMMENTED** [review]
Looks good to me for the operator side

File: .circleci/config.yml [hunk 1]
```diff
@@ -36,7 +36,7 @@ workflows:
 # https://circleci.com/blog/circleci-hacks-reuse-yaml-in-your-circleci-config-with-yaml/
 .defaults: &defaults
   docker:
-    - image: grafana/loki-build-image:0.23.0
+    - image: grafana/loki-build-image:0.24.0
```
Comment (chaudum) [inline .circleci/config.yml:39]: Shouldn't this be `0.24.1` as well?

**Review by chaudum — COMMENTED** [review]
One comment, but otherwise lgtm

File: .circleci/config.yml (hunk 1 above)
Comment (dannykopping) [reply .circleci/config.yml:39]: Good catch, thanks

**Review by chaudum — APPROVED** [review]

**Pull request merged**